target/linux/mediatek/image/mt7622.mk

```diff
@@ -37,6 +37,21 @@ define Build/bl31-uboot
 	cat $(STAGING_DIR_IMAGE)/mt7622_$1-u-boot.fip >> $@
 endef
 
+define Build/uboot-bin
+	cat $(STAGING_DIR_IMAGE)/mt7622_$1-u-boot.bin >> $@
+endef
+
+define Build/uboot-fit
+	$(TOPDIR)/scripts/mkits.sh \
+		-D $(DEVICE_NAME) -o $@.its -k $@ \
+		-C $(word 1,$(1)) \
+		-a 0x41e00000 -e 0x41e00000 \
+		-c "config-1" \
+		-A $(LINUX_KARCH) -v u-boot
+	PATH=$(LINUX_DIR)/scripts/dtc:$(PATH) mkimage -f $@.its $@.new
+	@mv $@.new $@
+endef
```

Comment:
These functions need to be moved to the main Makefile, since they are also needed for filogic targets.
Example: https://github.com/openwrt-xiaomi/openwrt/commit/f10db8873eab15606ea5d5c03f9a616187df0a8e

Function `uboot-fit` needs to be improved (made universal).
Example: https://github.com/openwrt-xiaomi/openwrt/commit/f10db8873eab15606ea5d5c03f9a616187df0a8e

Reply:
I don't think this kind of standalone utilities need to be included in the main OpenWrt tree. It can be built directly from mainline u-boot and live in a mainline u-boot fork instead.
BTW it'll be more complicated on filogic because the vendor bootchain expects UBI on top of NMBM. Our u-boot knows nothing about NMBM and it shouldn't be directly manipulating the UBI area.